eitprocessing/continuous_data/__init__.py

```diff
@@ -45,11 +45,14 @@ def __post_init__(self) -> None:
         self.lock("time")
 
     def __setattr__(self, attr: str, value: Any):  # noqa: ANN401
-        old_value = getattr(self, attr)
-        if isinstance(old_value, np.ndarray) and old_value.flags["WRITEABLE"] is False:
-            msg = f"Attribute '{attr}' is locked and can't be overwritten."
-            raise AttributeError(msg)
-        super().__setattr__(self, attr, value)
+        try:
+            old_value = getattr(self, attr)
+            if isinstance(old_value, np.ndarray) and old_value.flags["WRITEABLE"] is False:
+                msg = f"Attribute '{attr}' is locked and can't be overwritten."
+                raise ValueError(msg)
```

Comment:
Why is this a ValueError, and not an AttributeError? 

The description of AttributeError seems very appropriate for this case.

> Raised when an attribute reference (see [Attribute references](dfile:///Users/peter/Library/Application%20Support/Dash/DocSets/Python_3/Python%203.docset/Contents/Resources/Documents/doc/reference/expressions.html#attribute-references)) or assignment fails.

I think something like this would work. It catches a potential AttributeError from `getattr()`, but does allow us to raise an AttributeError ourselves. The `continue` statement should make it skip the `else` part.  


```suggestion
    try:
        old_value = getattr(self, attr)
    except AttributeError:
        continue
    else:
        if isinstance(old_value, np.ndarray) and old_value.flags['WRITEABLE'] is False:
            msg = "..."
            raise AttributeError(msg)
    super().__setattr__(attr, value)
```